generate.py

```diff
@@ -68,9 +68,11 @@ def init_scraper(class_name):
 
 
 def generate_test_data(class_name, host_name, content):
-    output = f"tests/test_data/{host_name}/{class_name.lower()}.testhtml"
-    with open(output, "w", encoding="utf-8") as target:
-        target.write(content.decode(encoding="utf-8"))
+    output = Path(f"tests/test_data/{host_name}/{class_name.lower()}.testhtml")
+    with output.open("w", encoding="utf-8") as target:
+        if isinstance(content, bytes):
+            content = content.decode("utf-8")
+        target.write(content)
```

Comment:
I still think it's preferable to keep the existing behaviour here, unless there's some reason to adjust it.

```suggestion
        target.write(content.decode(encoding="utf-8"))
```